sdk/iot/hub/samples/twin/src/iot_hub_twin_example.c

```diff
@@ -0,0 +1,447 @@
+      NULL));
+
+  return AZ_OK;
+}
+
+static void print_twin_response_type(
+    az_iot_hub_client_twin_response_type type,
+    MQTTClient_message* message)
+{
+  switch (type)
+  {
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
+      printf("A twin GET response was received\n");
+      if (message->payloadlen)
+      {
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+      }
+      break;
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+      printf("A twin desired properties message was received\n");
+      printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+      break;
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      printf("A twin reported properties message was received\n");
+      break;
+  }
+}
+
+static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
+{
+  (void)context;
+
+  if (topicLen == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
+  }
+
+  az_result result;
+
+  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
+
+  az_iot_hub_client_twin_response twin_response;
+  if (az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response) == AZ_OK)
+  {
+    printf("Twin Message Arrived\n");
+    print_twin_response_type(twin_response.response_type, message);
+    printf("Response status was %i\n", twin_response.status);
+  }
+
+  (void)result;
+
+  putchar('\n');
+  MQTTClient_freeMessage(&message);
+  MQTTClient_free(topicName);
+
+  return 1;
+}
+
+static int connect_device()
+{
+  int rc;
+
+  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
+  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+
+  char username[128];
+  size_t username_length;
+  if ((rc = az_iot_hub_client_get_user_name(&client, username, sizeof(username), &username_length))
+      != AZ_OK)
+
+  {
+    printf("Failed to get MQTT clientId, return code %d\n", rc);
+    return rc;
+  }
+
+  mqtt_connect_options.username = username;
+  mqtt_connect_options.password = NULL;
+
+  mqtt_ssl_options.keyStore = (char*)x509_cert_pem_file;
+  if (*x509_trust_pem_file != '\0')
+  {
+    mqtt_ssl_options.trustStore = (char*)x509_trust_pem_file;
+  }
+
+  mqtt_connect_options.ssl = &mqtt_ssl_options;
+
+  if ((rc = MQTTClient_connect(mqtt_client, &mqtt_connect_options)) != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to connect, return code %d\n", rc);
+    return rc;
+  }
+
+  return 0;
+}
+
+static int subscribe()
+{
+  int rc;
+
+  char twin_desired_topic[128];
+  if ((rc = az_iot_hub_client_twin_patch_get_subscribe_topic_filter(
+           &client, twin_desired_topic, sizeof(twin_desired_topic), NULL))
+      != AZ_OK)
+
+  {
+    printf("Failed to get C2D MQTT SUB topic filter, return code %d\n", rc);
```

Comment:
copy-paste issue: C2D here and a few other places (should be twin)

Reply:
fixed here and elsewhere